src/Repositories/SearchIndexRepository.php

```diff
@@ -314,25 +289,105 @@ function (Collection $threadsData) use ($now) {
                     ];
                 }
 
-                DB::connection(ConfigService::$databaseConnectionName)
-                    ->table(ConfigService::$tableSearchIndexes)
-                    ->upsert(
-                        $searchIndexes,
-                        ['thread_id', 'post_id']
-                    );
+                // perform the updateOrInsert inside a transaction, so we can push all of these in one transaction
+                // instead of causing A LOT of connections
+                DB::transaction(function () use ($searchIndexes) {
```

Comment:
I'm curious why this is much different that the previous logic.
Does the transaction force the connection to be reused where normally it creates a new one each time?

Reply:
The change to `updateOrInsert` was necessary to fix the bug where it's not finding the entries to update when trying to `upsert` instead, and `updateOrInsert` doesn't have an option to take an array of entries to update (or insert). As a result, this would lead to a separate database statement for every one of the searchIndexes in the array. By putting it into a transaction, it basically builds up all of the statements and runs them all in one big batch. Otherwise, there would be a new connection created each time (unless there's something deeper down in the framework to maintain the connection).